### PR TITLE
Feature/server and apis

### PR DESCRIPTION
**Pull request by rhyslbw**
Adds ContractProxy, ServiceApi, and Server models

### CODE REVIEW

File: src/server/application/ContractProxy.spec.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { expect } from 'chai'
+import * as http from 'http'
+import { listen, close } from '../lib/http'
+import { InMemoryRepository } from '../infrastructure/repositories'
```
Comment (Sam-Jeston) [inline src/server/application/ContractProxy.spec.ts]:
```suggestion
import { InMemoryRepository } from '../infrastructure'
```

File: src/server/application/Server.spec.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import { expect, use } from 'chai'
+import * as chaiAsPromised from 'chai-as-promised'
+import axios from 'axios'
+import { PubSub } from 'apollo-server'
+import { Server } from './Server'
+import { InMemoryRepository } from '../infrastructure/repositories'
```
Comment (Sam-Jeston) [inline src/server/application/Server.spec.ts]:
```suggestion
import { InMemoryRepository, HttpTarGzBundleFetcher, StubEngineClient  } from '../infrastructure'
```

File: src/server/application/ServiceApi.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        }
+      },
+      Mutation: {
+        loadContract (_: any, { contractAddress, bundleLocation }: { contractAddress: string, bundleLocation: string }) {
+          return contractController.load(contractAddress, bundleLocation)
+        }
+      },
+      Subscription: {
+        transactionSigningRequest: {
+          subscribe: (_: any, { publicKey }: { publicKey: string }) => pubSubClient.asyncIterator(`${Events.SIGNATURE_REQUIRED}.${publicKey}`)
```
Comment (Sam-Jeston) [inline src/server/application/ServiceApi.ts:53]: So you landed on leaving this in here for now?

That makes sense, particularly as we approach demo. Will be easy to move once things stabilise

File: src/server/application/ServiceApi.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        }
+      }
+    },
+    introspection: true
+  })
+  apolloServer.applyMiddleware({ app })
```
Comment (Sam-Jeston) [inline src/server/application/ServiceApi.ts:59]: What's the purpose of this?

I see you are returning the express instance. Is this only to adhere to the http.Server type for shutdown?

If so, according to the docs (https://www.apollographql.com/docs/apollo-server/api/apollo-server) new ApolloServer().listen() returns a promise, that when resolved is an object with a server key that is of the http.Server type

File: src/server/application/ContractProxy.spec.ts (hunk 1 above)
Comment (Sam-Jeston) [reply src/server/application/ContractProxy.spec.ts]: These infrastructure imports are getting a bit messy. I think it makes sense to export them from an index.ts file

**Review by Sam-Jeston — CHANGES_REQUESTED** [review]

File: src/server/application/ServiceApi.ts (hunk 3 above)
Comment (rhyslbw) [reply src/server/application/ServiceApi.ts:53]: Yes, it's a separate body of work, and I wanted to maintain the integrity of the integration test

File: src/server/application/ServiceApi.ts (hunk 4 above)
Comment (rhyslbw) [reply src/server/application/ServiceApi.ts:59]: Just added annotations to explain this

**Review by Sam-Jeston — APPROVED** [review]

**Pull request merged**